contracts/modules/v2/StakingPool.sol

```diff
@@ -446,85 +439,323 @@ contract StakingPool is IStakingPool, ERC721 {
   }
 
   function allocateStake(
-    uint productId,
-    uint period,
-    uint gracePeriod,
-    uint productStakeAmount,
-    uint rewardRatio
-  ) external onlyCoverContract returns (uint newAllocation, uint premium) {
+    CoverRequest calldata request
+  ) external onlyCoverContract returns (uint allocatedAmount, uint premium) {
 
     updateTranches();
 
-    Product memory product = products[productId];
-    uint allocatedProductStake = product.allocatedStake;
-    uint currentBucket = block.timestamp / BUCKET_DURATION;
+    // process expirations
+    uint gracePeriodExpiration = block.timestamp + request.period + request.gracePeriod;
+    uint firstTrancheIdToUse = gracePeriodExpiration / TRANCHE_DURATION;
```

Comment:
is this right? 

the name firstTrancheId makes me think of the first tranche cronologically but it takes the one matching the expiration time (last tranche)

Reply:
discusses offline, concluded it's correct